flink-connectors/flink-connector-kafka-0.9/src/main/java/org/apache/flink/streaming/connectors/kafka/internal/KafkaConsumerThread.java

```diff
@@ -151,6 +154,7 @@ public void run() {
 		// including concurrent 'close()' calls.
 		try {
 			this.consumer = getConsumer(kafkaProperties);
+			this.hasAssignedPartitions = !consumer.assignment().isEmpty();
```

Comment:
Can't we just start with `false` here?
We'll only ever get partitions once we enter the main fetch loop.